x/fswap/keeper/keeper.go

```diff
@@ -83,6 +82,12 @@ func (k Keeper) Swap(ctx sdk.Context, addr sdk.AccAddress, fromCoinAmount sdk.Co
 	return nil
 }
 
+func (k Keeper) CalcSwap(swapRate sdk.Dec, fromCoinAmount sdk.Int) sdk.Int {
+	multipliedAmountDec := swapRate.Mul(sdk.NewDecFromBigInt(fromCoinAmount.BigInt())).TruncateInt()
```

Comment:
Why not `return swapRate.MulTruncate(sdk.NewDecFromInt(fromCoinAmount)).TruncateInt()`